src/activate_linux.c

```diff
@@ -268,7 +290,16 @@ int main(int argc, char *argv[]) {
     overlay_width *= scale;
     verbose_printf("Scaled width:  %d px\n", overlay_width);
 
+    if(display_screen == 0)
+        display_screen = 0xFFFFFFFF;
+
     for (int i = 0; i < num_entries; i++) {
+        if(!(display_screen & 1 << i))
+        {
+            verbose_printf("Screen %i disabled\n", i);
+            continue;
```

Comment:
There may be case when no one screen matches chosen mask. I suggest `int` variable like `screens_with_overlays` incrementing each time overlay creates. And if no one overlay was created, terminate program with corresponding message.

Reply:
As the screen numbers are consecutive we can simply check if the N lower bits of the mask have at least one that is set, and exit if none are. Like so : if(!(display_screen & 0xFFFFFFFF >> (32 - num_entries)))

Reply:
@bourquenoud absolutely! Nice solution. But can you add parenthesis for `>>` , please :) Not all programmers have in their head order of operators